vm/systemSmartContracts/auction.go

```diff
@@ -137,17 +137,21 @@ func (s *stakingAuctionSC) unJail(args *vmcommon.ContractCallInput) vmcommon.Ret
 	totalUnJailPrice := big.NewInt(0).Mul(config.UnJailPrice, big.NewInt(int64(len(args.Arguments))))
 
 	if totalUnJailPrice.Cmp(args.CallValue) != 0 {
+
+		s.eei.SetReturnMessage("insufficient funds sent for unJail")
 		return vmcommon.UserError
 	}
 
 	err := s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.UnJail * uint64(len(args.Arguments)))
 	if err != nil {
+		s.eei.SetReturnMessage("insufficient gas limit")
 		return vmcommon.OutOfGas
 	}
 
 	for _, argument := range args.Arguments {
 		vmOutput, _ := s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(argument)))
 		if vmOutput != nil && vmOutput.ReturnCode == vmcommon.OutOfGas {
```

Comment:
there is not more insufficient gasLeft comming from stakingSC - all gas is taken by auction SC. verification here should be if err != nil then s.eei.AddReturnMessage(err.error()) continue. for vmOutput return message should be added to currently existing messages.

Reply:
done.